### PR TITLE
fix import from hash dialog

### PR DESCRIPTION
**Pull request by 0x6431346e**
## What changed?
It turns out the peers do load eventually (it's just very slow), but while testing the dialog I found out some other bug: If you change the hash, it no longer tries to get the peers and size, so I move the whole method in the store and called it after the text changed and is valid. 

This fixes: https://gitlab.com/siderus/Orion/issues/2

### CODE REVIEW

File: app/windows/Import/Components/Form.jsx [hunk 1]
```diff
@@ -20,7 +20,8 @@ class Form extends React.Component {
     if (isValid) {
       this.props.statsStore.hash = hash
       this.props.statsStore.isValid = isValid
-      this.forceUpdate()
+      this.props.statsStore.check()
+        .then(() => { this.forceUpdate() })
```
Comment (koalalorenzo) [inline app/windows/Import/Components/Form.jsx]: Should we add a catch here?

File: app/windows/Import/Stores/Stats.js [hunk 2]
```diff
@@ -31,6 +34,33 @@ export class StatStorage {
...
+        this.peersAmount = peers.length
+        this.isLoading = false
+      })
+
+    // Prepare the promise to check the stats
+    const pStats = getObjectStat(this.hash)
+      .then(stats => {
+        console.log('got stats')
+        this.stats = stats
+        this.isLoading = false
```
Comment (koalalorenzo) [inline app/windows/Import/Stores/Stats.js]: What about concurrency?
Should we have a `isLoading` both here and on line 46, or should we have a `.then` right after the `Promise.all`? What are the pro/cons of this?

File: app/windows/Import/Components/Form.jsx (hunk 1 above)
Comment (0x6431346e) [reply app/windows/Import/Components/Form.jsx]: The catch is in the stats store, shows an error dialog

File: app/windows/Import/Stores/Stats.js (hunk 2 above)
Comment (0x6431346e) [reply app/windows/Import/Stores/Stats.js]: I agree, must better to handle them in `Promise.all`

**Comment by koalalorenzo** [comment]
The behaviour now is a little bit slower:

The "find peers" is a little bit slower than the one that is fetching the size. Before it was loading them asynchronously, now instead we have to wait for both of them to be ready before showing the value.

What do you think of this change?

**Comment by 0x6431346e** [comment]
Hmm, I'll make them resolve async, getting the peers takes way too long

**Comment by 0x6431346e** [comment]
I've reverted the last commit, now they should resolve async, there no concurrency problem though, when one resolves before the other, the other will show `Loading...`

**Pull request merged**